core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -0,0 +1,290 @@
+
+var explorationEditorUrl = 'Exploration editor not loaded';
+var collectionEditorUrl = 'Collection editor not loaded';
+var topicEditorUrl = 'Topic editor not loaded';
+var storyEditorUrl = 'Story editor not loaded';
+var skillEditorUrl = 'Skill editor not loaded';
+
+var createButtonSelector = '.protractor-test-create-activity';
+var createCollectionButtonSelector = '.protractor-test-create-collection';
+
+var createTopicButtonSelector = '.protractor-test-create-topic-button';
+var topicNameField = '.protractor-test-new-topic-name-field';
+var topicUrlFragmentField = '.protractor-test-new-topic-url-fragment-field';
+var topicDescriptionField = '.protractor-test-new-topic-description-field';
+var topicThumbnailButton = '.protractor-test-photo-button';
+var topicUploadButton = '.protractor-test-photo-upload-input';
+var topicPhotoSubmit = '.protractor-test-photo-upload-submit';
+var thumbnailContainer = '.protractor-test-thumbnail-container';
+var confirmTopicCreationButton =
+  '.protractor-test-confirm-topic-creation-button';
+var createdTopicLink = '.protractor-test-topic-name';
+
+var createStoryButtonSelector = '.protractor-test-create-story-button';
+var storyNameField = '.protractor-test-new-story-title-field';
+var storyUrlFragmentField = '.protractor-test-new-story-url-fragment-field';
+var storyDescriptionField = '.protractor-test-new-story-description-field';
+var storyThumbnailButton = '.protractor-test-photo-button';
+var storyUploadButton = '.protractor-test-photo-upload-input';
+var storyPhotoSubmit = '.protractor-test-photo-upload-submit';
+var thumbnailContainer = '.protractor-test-thumbnail-container';
+var confirmStoryCreationButton =
+  '.protractor-test-confirm-story-creation-button';
+
+var createSkillButtonSelector = '.puppeteer-test-add-skill-button';
+var skillDescriptionField = '.protractor-test-new-skill-description-field';
+var skillOpenConceptCard = '.protractor-test-open-concept-card';
+var skillEditor = '.protractor-test-concept-card-text';
+var confirmSkillCreationButton =
+  '.protractor-test-confirm-skill-creation-button';
+var skillReviewMaterialInput = '.oppia-rte';
+
+const login = async function(browser, page) {
+  try {
+    // eslint-disable-next-line dot-notation
+    await page.goto(
+      ADMIN_URL, { waitUntil: networkIdle});
+    await page.waitForSelector('#admin', {visible: true});
+    await page.click('#admin');
+    await page.click('#submit-login');
+    // Checks if the user's account was already made.
+    try {
+      await page.waitForSelector('#username', {visible: true});
+      await page.type('#username', 'username1');
+      await page.click('#terms-checkbox');
+      await page.waitForSelector('#signup-submit');
+      await page.click('#signup-submit');
+      await page.waitForSelector('.oppia-navbar-dropdown-toggle');
+    } catch (error) {
+      // Already Signed in.
+    }
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log('Login Failed');
```

Comment:
Why do you need a console log?

Reply:
@nithusha21  Hi! So a brief explanation to these changes. I added a new folder in core/tests titled puppeteer. Inside the folder is a puppeteer script titled 'lighthouse_setup.js' which helps prepare certain pages to be tested by the lighthouse. 

I use console.log here to output error messages if the puppeteer script ever fails.